File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    sort(participant.begin(),participant.end());
    sort(completion.begin(),completion.end());
    
    int i=0;
    for( ; i < completion.size();i++){
        if(participant[i]!=completion[i]){
            answer= participant[i];
            break;
        }
    }
    return participant[i];
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        
        
        
        Arrays.sort(participant);
        Arrays.sort(completion);
        
        String answer = "";
        
        Boolean flag = false;
        
        for(int i = 0; i < completion.length; i++ ){
            if(!participant[i].equals(completion[i])){
                answer = participant[i];
                flag = true;
                break;
            }
        }
        
        if(flag == false){
            return participant[participant.length-1];
        }else{
            return answer;
        }
        
        
        
        
        
    }
}